dataprofiler/data_readers/data_utils.py

```diff
@@ -286,26 +314,27 @@ def read_csv_df(
     if len(selected_columns) > 0:
         args["usecols"] = selected_columns
 
-    # account for py3.6 requirement for pandas, can remove if >= py3.7
-    is_buf_wrapped = False
     if isinstance(file_path, BytesIO):
         # a BytesIO stream has to be wrapped in order to properly be detached
         # in 3.6 this avoids read_csv wrapping the stream and closing too early
         file_path = TextIOWrapper(file_path, encoding=encoding)
-        is_buf_wrapped = True
 
     fo = pd.read_csv(file_path, **args)
     data = fo.read()
 
     # if the buffer was wrapped, detach it before returning
-    if is_buf_wrapped:
+    if isinstance(file_path, TextIOWrapper):
```

Comment:
does this mean we can remove entire if statement though?
I'm concerned the statement means not just the is_wrapped statement.

Can we leave the code as is till we understand and add an issue to GH to address / remove?

Reply:
Or rather it looks like the code still achieves the same thing, but if we do this, we should at the very least not remove the comment from line 289

Reply:
I can add that comment right before the if statement on line 326.

Reply:
Actually for now, I can leave the original code and just cast to TextIOWrapper to satisfy mypy.